### PR TITLE
TDL-13115: Added support for checking credentials during discovery mode

### PR DESCRIPTION
**Pull request by harshpatel4crest**
# Description of change
TDL-13115: do_discover should check permissions of the token given

# Manual QA steps
 - Intentionally made some credentials with limited permissions and checked if the tap throws error.
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_github/__init__.py [hunk 1]
```diff
@@ -211,7 +214,50 @@ def get_catalog():
...
+        raise e
+    except BadCredentialsException as e:
+        logger.error("API token is invalid. Please enter correct credentials.")
+        raise e
+
+def verify_org_access(url_for_org, org):
+    try:
+        authed_get("verifying org access", url_for_org)
+    except NotFoundException as e:
+        logger.error("'%s' is not an Oragnization.", org)
```
Comment (karanpanchal-crest) [inline tap_github/__init__.py:231]: @harshpatel4crest There is a typo in the word Organization

Comment (harshpatel4crest) [reply tap_github/__init__.py:231]: Ok, I will solve that

File: tap_github/__init__.py [hunk 2]
```diff
@@ -99,25 +131,40 @@ def get_bookmark(state, repo, stream_name, bookmark_key):
...
+            raise exc(message) from None
+
+        except (ValueError, TypeError):
+            raise GithubException(error) from None
+
 # pylint: disable=dangerous-default-value
 def authed_get(source, url, headers={}):
     with metrics.http_request_timer(source) as timer:
         session.headers.update(headers)
         resp = session.request(method='get', url=url)
```
Comment (savan-chovatiya) [inline tap_github/__init__.py:158]: Add raise_for_error() here instead of every function which call authed_get function.

Comment (harshpatel4crest) [reply tap_github/__init__.py:158]: @savan-chovatiya Done the changes

**Comment by KAllan357** [comment]
Could you resolve the conflicts in this?

**Review by KAllan357 — APPROVED** [review]

**Pull request merged**